koku/api/query_handler.py

```diff
@@ -298,3 +298,44 @@ def _get_filter(self, delta=False):
         filters.add(query_filter=end_filter)
 
         return filters
+
+    def filter_to_order_by(self, parameters):
+        """Remove group_by[NAME]=* and replace it with group_by[NAME]=X.
+
+        The parameters object contains a list of filters and a list of group_bys.
+
+        For example, if the parameters object contained the following:
+        group_by[X] = Y
+        group_by[Z] = *     # removes this line
+        filter[Z] = L
+        filter[X] = Y
+
+        The returned parameters object would contain lists that look like this:
+
+        group_by[X] = Y
+        group_by[Z] = L     # adds this line
+        filter[Z] = L
+        filter[X] = Y
+
+        Thereby removing the star when there is a filter provided.
+
+        Args:
+            parameters (QueryParameters): The parameters object
+
+        Returns:
+            parameters (QueryParameters): The parameters object
+
+        """
+        for group_by_key in parameters.parameters['group_by']:
+            for group_by_value in parameters.parameters['group_by'][group_by_key]:
```

Comment:
Similar dictionary access comment as above:
`for group_by_value in parameters.parameters.get('group_by', {}).get(group_by_key):`

Reply:
If group_by is not in the parameters wouldn’t we want an error to be thrown?
https://github.com/project-koku/koku/blob/master/koku/api/query_params.py#

Reply:
Renamed

Reply:
added